go/vt/vttablet/tabletmanager/replication_reporter.go

```diff
@@ -31,6 +31,10 @@ import (
 
 var (
 	enableReplicationReporter = flag.Bool("enable_replication_reporter", false, "Register the health check module that monitors MySQL replication")
+	// neverChangeReplicationMaster can be set to false to prevent Vitess from ever updating the replication topology.
+	// This is useful if your replication topology is managed entirely outside of Vitess.
+	// Replication will still be started/stopped at the appropriate times.
+	neverChangeReplicationTopology = flag.Bool("never_change_replication_topology", false, "if set, vitess will never make changes to the replication topology, only stop and start slaves at the appropriate times.")
```

Comment:
The comment here should mention replication_reporter, like:
if set, the replication_reporter module will never make changes to the replication topology, only stop and start slaves at the appropriate times. Only used if -enable_replication_reporter flag is set.

Reply:
It doesn't matter that much but I disagree. I'm not sure we need to mention the `enable_replication_reporter` flag or the `replication_reporter` here.

This flag should make Vitess *never* update replication topology regardless if the `replication_reporter` is enabled or not. The fact that Vitess currently only tries to change replication topology if the `replication_reporter` is enabled is really irrelevant to a user that doesn't know exactly how Vitess works. Even I didn't understand that and I know Vitess better than many.

Also, if we ever build another feature that updates the replication topology this flag should *still* apply, even if the `enable_replication_reporter` flag is off.

Reply:
I see what you mean. We can chose to protect individual features, and then we have one flag per feature, and it's annoying. Or we can chose to have a global flag, and check it in all the places that use it. My comment was assuming the first choice, but it may be wrong.

For the record, we already have a similar flag in vtctl:
https://github.com/youtube/vitess/blob/master/go/vt/vtctl/reparent.go#L39
DisableActiveReparents = flag.Bool("disable_active_reparents", false, "if set, do not allow active reparents. Use this to protect a cluster using external reparents.")
It disables the Reparent methods. This seems similar here, so should we factor out that flag?

One way to do it would be to move the existing vtctl flag to a more common place (like mysqlctl), and use it directly here, instead of the new flag you added. Would that work?